src/python/pants/backend/codegen/protobuf/lint/buf/lint_rules.py

```diff
@@ -39,47 +38,51 @@ class BufLintRequest(LintTargetsRequest):
     name = "buf-lint"
 
 
-@rule(desc="Lint with buf lint", level=LogLevel.DEBUG)
-async def run_buf(request: BufLintRequest, buf: BufSubsystem) -> LintResults:
+@rule
+async def partition_buf(
```

Comment:
I hold hope we can find a way to make this not needed to be defined if the user wants this implementation.

The challenge is getting the subsystem and knowing the skip option.

Reply:
Cc @sureshjoshi